src/iexamine.cpp

```diff
@@ -2600,6 +2600,12 @@ void iexamine::fireplace( player &p, const tripoint &examp )
     const bool has_bionic_firestarter = p.has_bionic( bionic_id( "bio_lighter" ) ) &&
                                         p.power_level >= bionic_id( "bio_lighter" )->power_activate;
 
+    auto firequenchers = p.items_with( []( const item & it ) {
+        return it.damage_melee( DT_BASH );
+    } );
+
+    const bool has_firequencher = !firequenchers.empty();
```

Comment:
No need to declare this variable - in case of `!already_on_fire` below, on line 2612, this variable wouldn't be used at all.
```suggestion
```